### DISCUSSION TITLE
Banana Pi m2 Zero ?

### DISCUSSION THREAD

**Discussion by grandgooroo** [onlaj/Piano-LED-Visualizer#444] [Q&A]
Hi ! Nice project.

Did anyone manage to launch the image on a Banana Pi m2 Zero ?
I have no display on the HDMI output and nothing on the LCD. I tested the ISO (Armbian_21.08.1_Bananapim2zero_bullseye_current_5.10.60.img) which works. So it's not the card...

**Comment by onlaj** [comment]
Hey,
I'm pretty sure the preconfigured image won't work; you'll have to do a manual installation.
I have ordered one, and it should arrive this week. I'll be testing it then.

**Comment by grandgooroo** [comment]
Ok, thx, I'll try the manual installation, thanks !

**Comment by onlaj** [comment]
TL;DR: It probably won't work without some code modifications.

My Banana Pi arrived, but I couldn't find any "lite" system for that device. Additionally, it seems that preparing everything without a monitor and keyboard is not possible as saving wpa_supplicant.conf on the SD card doesn't work.

I installed [last year's version of Armbian](https://wiki.banana-pi.org/Banana_Pi_BPI-M2_ZERO#Basic_Development), connected a monitor and keyboard, and configured the WiFi. From that point on, I could connect to it through SSH.
I proceeded similarly to how I did with Raspberry Pi until I reached the step of "enabling SPI interface." At this point, it appeared slightly different, and I had to use the following command:
`sudo armbian-config `
Then I navigated to: system -> hardware -> enable spi-spidev.

When I tried to install the required packages, I encountered some errors. It seems that RPi.GPIO is built specifically for Raspberry Pi. The installation of rpi-ws281x also failed.

I suppose I can make it work with Banana Pi. However, considering that Raspberry Pi is going back in stock, I don't think it's worth the struggle.

**Reply by NikoTumi** [reply]
I have an orange pi and I had to use `pip install OPi.GPIO`
I'm still trying to get this setup to work with it.

**Comment by grandgooroo** [comment]
On my side I installed this image "2020-04-10-raspbian-jessie-ap6212-bpi-m2z-sd-emmc.img" and I managed
[syslog.txt](https://github.com/onlaj/Piano-LED-Visualizer/files/11509486/syslog.txt)
 to install everything, but I have an error when launching "visualizer.service":
"sudo systemctl start visualizer.service
Failed to start visualizer.service: Unit visualizer.service is not loaded properly: Invalid argument.
See system logs and 'systemctl status visualizer.service' for details."

I managed to find a cheap Raspberry zero, but until it arrives I would like to run the script on the banana ^^.

**Reply by onlaj** [reply]
I'll give it a try with raspbian tomorrow.

**Comment by onlaj** [comment]
I tried Raspbian and I get this error:

> from RPi._GPIO import *
> 
> RuntimeError: This module can only be run on a Raspberry Pi!

which is pretty straightforward.

So I installed ported version as stated [here](https://wiki.banana-pi.org/Getting_Started_with_M3#RPi.GPIO)
Now I get error:

```
  File "/home/Piano-LED-Visualizer/lib/neopixel.py", line 106, in begin
    raise RuntimeError('ws2811_init failed with code {0} ({1})'.format(resp, message))
RuntimeError: ws2811_init failed with code -3 (Hardware revision is not supported)
```

Updating that library doesn't help, I wonder if there is some fork for that.

Anyway, I still think it's better to just get Raspberry.

**Comment by grandgooroo** [comment]
Ok, thx. So i will wait to receive the raspberry ;).

**Comment by NikoTumi** [comment]
I tried it on a OrangePi Zero 2W and got the same issues
